package com.kelsonthony.maratonajava.javacore.Oexception.runtime;

public class RuntimeExceptionTest02 {

    public static void main(String[] args) {
        try {
            divisao(1, 0);
        } catch (RuntimeException e) {
            e.printStackTrace();
        }
        System.out.println("C처digo finalizado");
    }

    /**
     * @param a
     * @param b n찾o pode ser zero
     * @return int
     * @throws IllegalArgumentException caso b seja zero
     */
    private static int divisao(int a, int b) {
        if (b == 0) {
            throw new IllegalArgumentException("N찾o aceita divis찾o por zero");
        }
       /* try {
            return a / b;
        }catch (RuntimeException e) {
            e.printStackTrace();
        }
        return 0;*/
        return a / b;
    }
}
